tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -1587,6 +1587,10 @@ const unsigned char *sqlite3_value_text(sqlite3_value *pVal) {
 	}
 
 	if (pVal->type == SQLiteTypeValue::TEXT || pVal->type == SQLiteTypeValue::BLOB) {
+		// check if the string has already been allocated
+		if (pVal->szMalloc > 0) {
+			return (const unsigned char *)pVal->zMalloc;
```

Comment:
The same problem would still occur if we called `sqlite3_value_text` twice on a `SQLiteTypeValue::INTEGER` or `SQLiteTypeValue::FLOAT`, right?  Could you add code that fixes it in that case as well + add a test case for that? Thanks!

Reply:
Good catch, I will add a new test case for that. Thanks!